woocommerce/woocommerce-ios
# [Subscriptions in Order Creation 3] Update product selector

Closes #12435
Closes #12436


## Description
This PR updates the product rows to add support for product subscriptions, this includes:
- Simple subscriptions, variable subscriptions and its variations
- Pricing and frequency
- Signup fees, and free trial periods

| Simple Subscription | Variable Subscription | Subscription Variations |
|--------|--------|--------|
| ![20240409-subscriptions UI 1 simple subs](https://github.com/woocommerce/woocommerce-ios/assets/3812076/d4615381-c344-429a-aab5-a46bf57c5357) | ![20240409-subscriptions UI 2 variable subs](https://github.com/woocommerce/woocommerce-ios/assets/3812076/9a6e31f1-5271-447d-a815-9d68952ce74a) | ![20240409-subscriptions UI 3 variations](https://github.com/woocommerce/woocommerce-ios/assets/3812076/b9e6d42a-4a10-4822-902c-0729e17b54f9) | 

## Caveats
The API/wp-admin data validation seems to accept data that I would consider odd for subscriptions, for example both empty and zero signup fees, 0-day trial periods, etc, ...). In order to cover these cases UI-wise I've attempted some triangulation for all labels and handled cases for `0`, `empty`, `one`, and `some` values in order to display meaningful data.

In general, we should display nothing for trials or fees that happen to be empty or zero. Happy to iterate further once we add backend support on the sub-projects that will run in parallel, since we might be able to improve this directly in the API, but I don't think is worth too much additional time right now.

## Changes
- Updated the `ProductRowViewModel` with a new `ProductSubscription` property, which represents the subscription settings extracted from product meta data for a Subscription-type Product (if any).
- With this, we update the different initializers for product rows to account for this data.
- We format the output as per the proposed designs ( Q9tkFMtymZz0yhfLLbKhig-fi-2176_61285 ). Since we only get the raw values from the API we have to format the different properties to account for currency, frequency, localization, etc, ... 

## Testing instructions
* On a store using WooCommerce Subscriptions, create a simple subscription, a variable subscription, and variations. You can use the testing site https://indiemelon.mystagingwebsite.com/ if you prefer, that has already these setup.
* Go to Orders > `+` > search for "daily virtual" and observe that:
  * Product Subscriptions with no sign up fees or trials will only show the billing frequency
  * Product Subscriptions with sign up fees and/or trials will only these, as well as the billing frequency
* Search for "variable" (2nd screenshot) observe that upon tapping on the variable subscription product, the subscription variations will show appropriately as well (3rd screenshot)
* Set the `.subscriptionsInOrderCreationUI` feature flag to false and re-run the app
* Go back to Orders > `+` and observe that subscription products  do not show subscription details now.
